src/HealthChecks/HealthChecks/src/Microsoft.Extensions.Diagnostics.HealthChecks.csproj

```diff
@@ -28,6 +30,10 @@ Microsoft.Extensions.Diagnostics.HealthChecks.IHealthChecksBuilder
     <Reference Include="Microsoft.Extensions.Hosting.Abstractions" />
     <Reference Include="Microsoft.Extensions.Options" />
     <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
+
+    <!-- Required for logging source generation -->
+    <!-- workaround https://github.com/dotnet/runtime/issues/52222 -->
+    <Reference Include="Microsoft.Extensions.Logging" PrivateAssets="All" />
```

Comment:
This shouldn't be necessary. The reference to Logging.Abstractions should suffice